src/index/DeltaTriples.cpp

```diff
@@ -118,32 +120,87 @@ DeltaTriplesCount DeltaTriples::getCounts() const {
   return {numInserted(), numDeleted()};
 }
 
+// _____________________________________________________________________________
+DeltaTriples::Triples DeltaTriples::makeInternalTriples(
+    const Triples& triples) {
+  Triples internalTriples;
+  ad_utility::util::LRUCache<Id::T, ad_utility::triple_component::Iri>
+      predicateCache{10};
+  for (const auto& triple : triples) {
+    const auto& ids = triple.ids();
+    Id objectId = ids.at(2);
+    auto optionalLiteralOrIri = ExportQueryExecutionTrees::idToLiteralOrIri(
+        index_, objectId, localVocab_, true);
+    if (!optionalLiteralOrIri.has_value() ||
+        !optionalLiteralOrIri.value().isLiteral() ||
+        !optionalLiteralOrIri.value().hasLanguageTag()) {
+      continue;
+    }
+    const auto& predicate =
+        predicateCache.getOrCompute(ids.at(1).getBits(), [this](Id::T bits) {
+          auto optionalPredicate = ExportQueryExecutionTrees::idToLiteralOrIri(
+              index_, Id::fromBits(bits), localVocab_, true);
+          AD_CORRECTNESS_CHECK(optionalPredicate.has_value());
+          AD_CORRECTNESS_CHECK(optionalPredicate.value().isIri());
+          return std::move(optionalPredicate.value().getIri());
+        });
+    auto specialPredicate = ad_utility::convertToLanguageTaggedPredicate(
+        predicate,
+        asStringViewUnsafe(optionalLiteralOrIri.value().getLanguageTag()));
+    Id specialId = TripleComponent{std::move(specialPredicate)}.toValueId(
+        index_.getVocab(), localVocab_, index_.encodedIriManager());
+    internalTriples.push_back(
+        IdTriple<0>{std::array{ids.at(0), specialId, ids.at(2), ids.at(3)}});
+  }
```

Comment:
1. If this function computes in the ID space, then the predicate cache can also be in the ID space (then we can afford it to be much larger, can't it)

2. This is (at least semantically) code duplication to stuff that is done in the `IndexBuilderTypes.h` , do we get these implementations closer to each other, (the types are different, but if they are in the same file, it is easier to keep track of `also add stuff here if you change something`.